server/src/main/java/org/elasticsearch/cluster/health/ClusterHealthStatus.java

```diff
@@ -64,4 +67,19 @@ public static ClusterHealthStatus fromString(String status) {
             throw new IllegalArgumentException("unknown cluster health status [" + status + "]");
         }
     }
+
+    public static ClusterHealthStatus fromClusterState(ClusterState clusterState, Index index) {
+        IndexRoutingTable indexRoutingTable = clusterState.routingTable().index(index);
```

Comment:
We could probably just pass in `IndexRoutingTable` here, since we don't need the `index` at all in the rest of the method.